single_kernel_mongo/core/operator.py

```diff
@@ -393,3 +393,10 @@ def build_local_tls_directory(self) -> None:
             return
 
         Path(self.state.paths.conf_path).mkdir(exist_ok=True)
+
+    def instantiate_keyfile(self):
+        """Instantiate the keyfile."""
+        if not (keyfile := self.state.get_keyfile()):
+            raise Exception("Waiting for leader unit to generate keyfile contents")
```

Comment:
this is okay for the current PR, as it's been simply relocated from elsewhere in the code, but we should scan those in the future to have more specific exceptions

Reply:
Agreed, I'll create an issue for that